pandas/core/reshape/merge.py

```diff
@@ -123,6 +123,9 @@
     np.object_: libhashtable.ObjectFactorizer,
 }
 
+if np.intc is not np.int32:
```

Comment:
Could you add the Github issue number as a comment here?

Reply:
```suggestion
# See https://github.com/pandas-dev/pandas/issues/52451
if np.intc is not np.int32:
```